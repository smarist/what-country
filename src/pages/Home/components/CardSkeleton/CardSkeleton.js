import React from 'react';
import { Skeleton } from '@material-ui/lab';
import { Grid } from '@material-ui/core';
import useStyles from './CardSkeleton.style';

function CardSkeleton() {
  const classes = useStyles();
  return (
    <Grid container data-testid="card-skeleton" className={classes.skeletonBox}>
      {new Array(50).fill('').map((item, index) => (
        <Skeleton
          key={`skeleton-${index}`}
          animation="wave"
          className={classes.skeleton}
        />
      ))}
    </Grid>
  );
}

export default CardSkeleton;
