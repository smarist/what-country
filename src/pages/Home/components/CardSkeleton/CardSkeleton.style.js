import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles(() => ({
  skeletonBox: {
    display: 'grid',
    gridGap: '70px',
    gridTemplateColumns: 'repeat(auto-fill, minmax(230px, 1fr))',
    gridAutoRows: '320px',
  },
  skeleton: {
    height: '430px',
    borderRadius: '3px',
    // maxWidth: '230px',
  },
}));

export default useStyles;
