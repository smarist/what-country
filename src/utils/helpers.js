export function objectToArray(val) {
  const obj = val;
  return Object.entries(obj);
}

let timeoutId;

export function debounceHandler(func, delay) {
  if (timeoutId) clearTimeout(timeoutId);
  timeoutId = setTimeout(func, delay);
}
