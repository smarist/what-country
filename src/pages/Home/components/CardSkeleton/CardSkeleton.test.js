import { render } from '@testing-library/react';
import { withThemeProvider } from '../../../../tests/helpers';
import CardSkeleton from './CardSkeleton';

describe('Card Skeleton Test', () => {
  it('should render successfully', () => {
    const { queryByTestId } = render(withThemeProvider(CardSkeleton, { value: '' }));
    const card = queryByTestId('card-skeleton');

    expect(card).toBeTruthy();
    expect(card).toMatchSnapshot();
  });
});
