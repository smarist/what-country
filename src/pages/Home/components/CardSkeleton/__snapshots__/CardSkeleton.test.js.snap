// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Skeleton Test should render successfully 1`] = `
<div
  class="MuiGrid-root makeStyles-skeletonBox-1 MuiGrid-container"
  data-testid="card-skeleton"
>
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
  <span
    class="MuiSkeleton-root MuiSkeleton-text makeStyles-skeleton-2 MuiSkeleton-wave"
  />
</div>
`;
